python/pyxbos/pyxbos/drivers/obvius/obvius.py

```diff
@@ -0,0 +1,198 @@
+                meter_path = building_path + '/' + self.to_pathname(metername)
+                req_url = url.format(starttime, endtime) \
+                    + "&mnuStartMonth=" + str(starttime.month) + "&mnuStartDay=" + str(starttime.day) \
+                    + "&mnuStartYear=" + str(starttime.year) + "&mnuStartTime=" + str(starttime.hour) + "%3A" + str(starttime.minute) \
+                    + "&mnuEndMonth=" + str(endtime.month) + "&mnuEndDay=" + str(endtime.day) \
+                    + "&mnuEndYear=" + str(endtime.year) + "&mnuEndTime=" + str(endtime.hour) + "%3A" + str(endtime.minute) \
+
+                response = requests.get(url=req_url, auth=self.auth)
+                if response:
+                    data = response.content.decode('utf-8')
+                    if "No data found within range" in data:
+                        print("No data found for "+meter_path+" within given date range")
+                        continue
+                    csv_df = pd.read_csv(io.StringIO(data))
+
+                    if "water" in meter_path:
+                        for index, row in csv_df.iterrows():
+                            rowDict = dict(row)
+                            meter_data = iot_pb2.Meter(
+                                water_total = types.Double(value=rowDict.get('Water (Gallons)')),
+                                water_rate = types.Double(value=rowDict.get('Water Ave Rate (Gpm)')),
+                                water_instantaneous = types.Double(value=rowDict.get('Water Instantaneous (Gpm)')),
+                            )
+                            msg = xbos_pb2.XBOS(
+                                XBOSIoTDeviceState = iot_pb2.XBOSIoTDeviceState(
+                                    time = int(time.time()*1e9),
+                                    meter = meter_data
+                                )
+                            )
+                            self.report(self.hostname, msg)
+
+                    if "condensate" in meter_path:
+                        for index, row in csv_df.iterrows():
+                            rowDict = dict(row)
+                            meter_data = iot_pb2.Meter(
+                                condense_total = types.Double(value=rowDict.get('Steam Condensate Meter (Gallons)')),
+                                condense_rate = types.Double(value=rowDict.get('Steam Condensate Meter Ave Rate (Gpm)')),
+                                condense_instantaneous = types.Double(value=rowDict.get('Steam Condensate Meter Instantaneous (Gpm))')),
+                            )
+                            msg = xbos_pb2.XBOS(
+                                XBOSIoTDeviceState = iot_pb2.XBOSIoTDeviceState(
+                                    time = int(time.time()*1e9),
+                                    meter = meter_data
+                                )
+                            )
+                            self.report(self.hostname, msg)
```

Comment:
Here and the other `self.report` calls should have a more substantial and descriptive resource, rather than just the hostname of the machine. What data do we have access to? Building/meter name?

Reply:
From in-person sync: Will replace hostname (this field will be deleted anyway) with simplified schema of "building/type of meter" i.e. "soda-hall/electric".

Reply:
Resolved, replaced instances of `self.hostname` with `building + '/<meter type>_meter`.